File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/molecules/header/MolSectionTitle.kt
package com.perrystreet.woof.designsystem.atomic.molecules.header

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.SpacingRoles
import com.perrystreet.woof.designsystem.atomic.atoms.divider.AtomHorizontalDivider
import com.perrystreet.woof.designsystem.atomic.atoms.text.AtomText
import com.perrystreet.woof.designsystem.atomic.atoms.text.roles.TextFontRole
import com.perrystreet.woof.designsystem.atomic.molecules.header.roles.SectionToneRole

@Composable
fun MolSectionTitle(
    title: String,
    toneRole: SectionToneRole = SectionToneRole.OnScrim,
) {
    Column(
        modifier = Modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(SpacingRoles.Component.Compact.dp),
    ) {
        AtomText(
            text = title,
            textFontRole = TextFontRole.DisplayH4,
            colorRole = toneRole.titleColorRole,
            maxLines = 1,
        )
        AtomHorizontalDivider()
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Molecules/Header/MolSectionTitle.swift
import SwiftUI

public struct MolSectionTitle: View {
    private let title: String
    private let toneRole: SectionToneRole

    public init(title: String, toneRole: SectionToneRole = .onScrim) {
        self.title = title
        self.toneRole = toneRole
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: SpacingRoles.Component.compact.rawValue) {
            AtomText(text: title, textFontRole: .displayH4, colorRole: toneRole.titleColorRole, maxLines: 1)
            AtomHorizontalDivider()
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}
